### DISCUSSION TITLE
Plans for dropping for OAS2 ?

### DISCUSSION THREAD

**Discussion by sergioisidoro** [rswag/rswag#503] [General]
Hi everyone, 

Right now supporting OAS2 adds quite a big overhead, and I'm wondering if it really makes sense to continue supporting it. OAS 3.0.0 was released on 26 July 2017, and if we to continue to keep this gem in good shape, it's time to focus our resources into supporting OAS >= 3

Some examples of the overhead in keeping backwards compatibility include recent pull requests, like:
* Response examples - https://github.com/rswag/rswag/pull/501
* Query - https://github.com/rswag/rswag/pull/499 

What do you think? Please react with 👎 if still need OAS2 support.

**Comment by lafferty151** [comment]
burn it out. It's difficult to maintain and it also leads to confusion as a user when trying to filter out the old conventions vs. the new.

**Reply by jtannas** [reply]
Agreed. We're a pretty small group maintaining this gem so supporting old versions saps energy better used elsewhere.
Maybe a `v2.5.2` release with a deprecation notice for OAS2 support so we can start the ball rolling?

**Comment by jtannas** [comment]
Vaguely related... This seems like it'd be a major version change (ie. `v3.0.0`) due to it breaking lots of projects.
Skimming through the PRs I saw #396 which would also require a major version change.

Now that we have more maintainers on the project, should we let loose a bit and do a major refactor for v3?

**Comment by sergioisidoro** [comment]
This has been up for 15 days now. I'm not sure if there's any other way to reach people, but so far it seems there's no negative responses. 

I think there's good points here about the refactoring / major release, but we need to be strategic with the amount of resources we have. A re-write, refactor would likely be too big of a bite, and it would likely end in failure. I'm more in favour of issuing a deprecation warning, and then gradually refactor things away from OAS2.

My biggest concern right now would be moving the `metadata` example object to a data structure closer to OAS3, and start go get rid of these `upgrade` methods: https://github.com/rswag/rswag/blob/05e22c3bd7fff53d40cd8104678dab73d0b7aff7/rswag-specs/lib/rswag/specs/swagger_formatter.rb#L38

**Reply by jtannas** [reply]
Sorry, been away from computers for a few weeks but I can start helping out again now.

> I think there's good points here about the refactoring / major release, but we need to be strategic with the amount of resources we have.

Good call. I'll focus on getting reviews done on existing PRs before starting any of my own work.

**Reply by romanblanco** [reply]
Starting this effort here: https://github.com/rswag/rswag/pull/574
